### PR TITLE
Pract.06: Add 01-02 | Nikita Beliy

### PR DESCRIPTION
**Pull request by N1k1ta-white**
Nikita Beliy
SI 1 9MI800022

### CODE REVIEW

**Comment by Justsvetoslavov** [comment]
@GeorgiTerziev02 @khatoni Can you take a look?

Edit: Task01 is an example solution from me, so maybe more towards the Task02.

File: Deck/Deck.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    void editMagic(const Magic& card, const int pos);
+
+    void editMonster(const Monster &card, const int pos);
+
+    void delMagic(const int pos);
+
+    void delMonster(const int pos);
+
+    Deck& operator=(Deck deck1);
```
Comment (khatoni) [inline Deck/Deck.h:28]: You do not need to overload operator = here

File: Duelist/Duelist.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    Duelist() {
+        name = new char[1];
+        name[0] = '\0';
+    }
+
+    Duelist(const char* name, Deck deck1);
+
+    ~Duelist() {
+        delete[] name;
+    }
```
Comment (khatoni) [inline Duelist/Duelist.h]: since this class has dynamic memory you need also copy constructor and copy assignment operator=

File: Item/Item.h [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    double getPrice() const;
+
+    void setName(const char* name);
+
+    void setAmount(const int amount);
+
+    void setPrice(const int price);
+
+    void copyItem(const Item& item);
```
Comment (khatoni) [inline Item/Item.h]: make this function private

File: Item/Item.h [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    Item& operator=(const Item& item);
+
+    void delItem();
```
Comment (khatoni) [inline Item/Item.h]: make this private as well

File: Monster/Monster.h [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    char* name;
+    unsigned int damage;
+    unsigned int hp;
+
+public:
+
+    Monster();
+
+    Monster(const char* name, const unsigned int enemies, const unsigned int defenders);
+
```
Comment (khatoni) [inline Monster/Monster.h:17]: No copy constructor

File: Monster/Monster.cpp [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+Monster &Monster::operator=(const Monster& monster) {
+    delete[] name;
+    name = new char[strlen(monster.name) + 1];
+    strcpy(this->name, monster.name);
+    this->hp = monster.hp;
+    this->damage = monster.damage;
+    return *this;
+}
```
Comment (khatoni) [inline Monster/Monster.cpp]: Where is the check if (this!=&other)

File: ShoppingCart/ShoppingCart.h [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+class ShoppingCart {
+    Item* items;
+    int size;
+    int count;
+
+public:
+    ShoppingCart();
+
+    ~ShoppingCart();
+
```
Comment (khatoni) [inline ShoppingCart/ShoppingCart.h:17]: BIG 4 ????

File: ShoppingCart/ShoppingCart.cpp [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+    Item temp(item1);
+    item1 = item2;
+    item2 = temp;
+}
+
+void ShoppingCart::addSpace() {
+    Item *copyItems = new Item[size];
+    for (int i = 0; i < size; ++i)
+        copyItems[i] = items[i];
+    delete[] items;
```
Comment (khatoni) [inline ShoppingCart/ShoppingCart.cpp]: this is redundant

**Review by khatoni — CHANGES_REQUESTED** [review]
There are some big mistakes

**Pull request closed without merging**